### PR TITLE
Add image push GitHub action for Prod

### PR DESCRIPTION
**Pull request by Stretch96**
## Changes

* Adds a GitHub action to push a specified release/tag to Azure Container Registry within the Production environment

## Checklist

- [ ] Attach this pull request to the appropriate card in Trello.
- [ ] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

File: .github/workflows/build-and-push-image-prod.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: Continuous Delivery / Prod
```
Comment (mec) [inline .github/workflows/build-and-push-image-prod.yml:1]: Could this be `Production` please.

File: .github/workflows/build-and-push-image-prod.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    name: Set Test Release Version
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{steps.release-version.outputs.version}}
+    steps:
+      - id: release-version
+        run: echo "::set-output name=version::release-7"
+
+  build-and-push-image-test:
+    name: Build and push image test
```
Comment (mec) [inline .github/workflows/build-and-push-image-prod.yml:19]: This should be `Production` as well, it's a small thing but potentially confusing.

File: .github/workflows/build-and-push-image-prod.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          fi
+          echo ::set-output name=tags::${TAGS}
+          echo ::set-output name=deploy-version::${VERSION}
+
+      - name: Push image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: ./Dockerfile
+          build-args: |
```
Comment (mec) [inline .github/workflows/build-and-push-image-prod.yml:55]: Why do we need `build-args` and why is the `RAILS_ENV` development? I've never understood.

**Review by mec — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-push-image-prod.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        id: prepare-tags
+        run: |
+          DOCKER_IMAGE=${{ secrets.PROD_ACR_URL }}/complete-app
+          VERSION=latest
+          TAGS="${DOCKER_IMAGE}:${VERSION}"
+          if [ "${{ github.event_name }}" = "push" ]; then
+            VERSION=sha-${GITHUB_SHA}
+            TAGS="$TAGS,${DOCKER_IMAGE}:${VERSION},${DOCKER_IMAGE}:${{needs.set-test-release-version.outputs.version}}"
+          fi
+          echo ::set-output name=tags::${TAGS}
```
Comment (mec) [inline .github/workflows/build-and-push-image-prod.yml:47]: `set-output` is deprecated - should we change this now?

**Review by mec — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-push-image-prod.yml (hunk 4 above)
Comment (Stretch96) [reply .github/workflows/build-and-push-image-prod.yml]: I've added a card to our Trello, we can change them all at once in a different PR 👍

**Pull request merged**